packages/next/src/server/web/spec-extension/unstable-cache.ts

```diff
@@ -39,7 +47,11 @@ export function unstable_cache<T extends Callback>(
         // Set during dynamic rendering
         store.revalidate = 0
         // If revalidate was already set in the store before we should check if the new value is lower, set it to the lowest of the two.
-      } else if (typeof store.revalidate === 'number') {
+      } else if (
+        typeof store.revalidate === 'number' &&
+        typeof options.revalidate === 'number' &&
+        !isNaN(options.revalidate)
+      ) {
```

Comment:
`options.revalidate` looks like it'd already be validated on line 37, but should we add `validateRevalidate(store.revalidate)`?

Reply:
This is validating the new value from options not the store value.